test/test.js

```diff
@@ -52,6 +64,16 @@ describe('Valid Input', function () {
         });
     });
 
+    it('Map', function () {
+        confirmWorks(new Map([['a', 1], [{}, 2], [{}, 5], [0, 3]]));
```

Comment:
Are these (and your other tests) actually working? This code doesn't throw an error:

    var x = new Map([['a', 1]])
    var y = new Map([['b', 1]])
    assert.deepEqual(x, y);

Maybe we need to switch to an assertion library that supports these new data types.